doc/doxygen/headers/automatic_and_symbolic_differentiation.h

```diff
@@ -20,8 +20,34 @@
  * to automatic and symbolic differentiation.
  *
  * Below we provide a very brief introduction as to what automatic and symbolic differentiation are,
- * what variations of these computational / numerical schemes exist, and how they are integrated
- * within deal.II's framework.
+ * what variations of these computational/numerical schemes exist, and how they are integrated
+ * within deal.II's framework. The purpose of all of these schemes is to automatically compute
+ * (approximations of) the derivative of functions in cases where one does not want to
+ * compute them by hand. Common examples are situations in the finite element context is where
+ * one wants to solve a nonlinear problem that is given by requiring that some residual
+ * $F(u,\nabla u)=0$ where $F$ is a complicated function that needs to be differentiated to
+ * apply a Newton method; and situations where one is given a parameter dependent problem
+ * ${\cal A}(q,u,\nabla u) = f$ and wants to form derivatives with regards to $q$, for example
+ * to optimize an output functional with regards to $q$, or for a sensitivity analysis with
```

Comment:
What does "q" represent here? I presume its the "parameter" (material, geometric or something else?) that you refer to earlier?